deployments.go

```diff
@@ -77,6 +77,34 @@ func (s *deployerService) Deploy(ctx context.Context, db *gorm.DB, opts DeployOp
 	if err := json.NewEncoder(opts.Output).Encode(&msg); err != nil {
 		return r, err
 	}
-
+	go s.streamEvents(opts)
 	return r, err
 }
+
+func (s *deployerService) streamEvents(opts DeployOpts) {
+	for event := range opts.Events {
+		var msg string
+		if event.Error != nil {
+			msg = fmt.Sprintf("Error: %s", event.Error.Error())
+		} else {
+			if event.Message != "" {
+				msg = fmt.Sprintf("Status: %s", event.Message)
+			}
+		}
+
+		if msg != "" {
+			m := jsonmessage.JSONMessage{Status: msg}
+			if err := json.NewEncoder(opts.Output).Encode(&m); err != nil {
+				close(opts.Events)
```

Comment:
right, that makes sense. i was under the impression whoever was consuming from the channel should be the one to close it, but this will make things cleaner